File: Data Structures & Algorithms/count-subsequences/submission-13.cs
public class Solution {
    public int NumDistinct(string s, string t) {
        if (t.Length > s.Length) {
            return 0;
        }
        return Dfs(s, t, 0, 0);
    }

    private int Dfs(string s, string t, int i, int j) {
        if (j == t.Length) {
            return 1;
        }
        if (i == s.Length) {
            return 0;
        }

        int res = Dfs(s, t, i + 1, j);
        if (s[i] == t[j]) {
            res += Dfs(s, t, i + 1, j + 1);
        }
        return res;
    }
}

File: Data Structures & Algorithms/coin-change-ii/submission-13.java
public class Solution {
    public int change(int amount, int[] coins) {
        Arrays.sort(coins);

        return dfs(coins, 0, amount);
    }

    private int dfs(int[] coins, int i, int a) {
        if (a == 0) {
            return 1;
        }
        if (i >= coins.length) {
            return 0;
        }

        int res = 0;
        if (a >= coins[i]) {
            res = dfs(coins, i + 1, a);
            res += dfs(coins, i, a - coins[i]);
        }
        return res;
    }
}